x-pack/plugins/security_solution/public/plugin.tsx

```diff
@@ -76,6 +76,7 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
   private appUpdater$ = new Subject<AppUpdater>();
 
   private storage = new Storage(localStorage);
+  private sessionStorage = new Storage(sessionStorage);
```

Comment:
Any specific reason we decided for using `sessionStorage` instead of `localStorage`? I can imagine a user closing a tab, or window, and reopening it, and still want the table to have its state persisted.

Reply:
Right, `I can imagine a user closing a tab, or window, and reopening it, and still want the table to have its state persisted.` this is the scenario which we want to avoid. The local storage persists the state for too long so returning back in a month the user will still have the same state persisted while the filtering and paging data may be outdated and does not show the same rules on the page. @xcrzx commented [here](https://github.com/elastic/kibana/pull/145111#discussion_r1023936654) on this and after the thorough discussion we decided that the session storage fits better.